src/lsm/table_iterator.zig

```diff
@@ -39,9 +39,6 @@ pub fn TableIteratorType(comptime Table: type, comptime Storage: type) type {
         /// out of blocks in the blocks ring buffer but haven't buffered a full block of
         /// values in memory. In this case, we copy values from the head of blocks to this
         /// ring buffer to make that block available for reading further values.
-        /// Thus, we guarantee that iterators will always have at least a block's worth of
-        /// values buffered. This simplifies the peek() interface as null always means that
-        /// iteration is complete.
```

Comment:
Isn't the first sentence deleted still true?

Reply:
I wasn't too sure about how much exactly was guaranteed to be buffered. Just knew that at some points, no value was buffered while the iterator still wasn't finished (so went pessimistic). I believe that invariant is still true though due to the block coalescing logic below.

Reply:
Yeah, I think the first sentence is still true, though the second is false.